src/main/ts/Part2Ex4FP.ts

```diff
@@ -149,17 +157,19 @@ Again, this looks familiar from our getOrElse1 function above.
 
 TODO: rewrite getOrElse1 using both Fun.identity and the "constant" function defined above.
  */
-
+export const getOrElse3 = <A> (oa: Optional<A>, other: A): A => oa.fold(constant(other), Fun.identity)
 
 /*
 TODO: use katamari's Fun.constant in your getOrElse and see if it compiles.
  */
+export const getOrElse4 = <A> (oa: Optional<A>, other: A): A => oa.fold(Fun.constant(other), Fun.identity)
 
 // TODO: Write a function that takes an array of numbers and replaces each value with 9.
+export const replaceElementWith9 = (items: number[]) => Optional.from(items).fold(Fun.constant([]), (items) => items.map(Fun.constant(9)));
```

Comment:
This is overly complicated, as again `items` shouldn't be nullable so you don't need the `Optional.from` logic here. Additionally as a heads up we generally avoid the native `Array.map` function and instead use the helper functions defined in `Arr`. So instead this could have been something like this:
```suggestion
export const replaceElementWith9 = (items: number[]) => Arr.map(items, Fun.constant(9));
```